### ISSUE TITLE
Remote debugging not working on android

### ISSUE THREAD

**Issue by hedgepigdaniel** [callstack/haul#405]
### Current Behavior

When starting an app with the setting "Debug JS Remotely" enabled, a RSOD appears with a java error which reads "unexpected url: /haul-hmr?platform=android"

I think the bug was introduced here: https://github.com/callstack/haul/pull/387/files

### Expected Behavior

Remote debugging should not stop the code from working

### Your Environment

React Native debugger 0.7.18
haul: https://github.com/hedgepigdaniel/haul/tree/b06ffb3bb84803ed08435eb3363a2468fee1da31 (based off recent master)
OS: Arch linux
OS: Android 8.1

From the console in react native debugger (the value of window.location does not seem to match the expectation in `polyfillEnvironment.js`.
```
window.location
WorkerLocation {href: "file:///opt/react-native-debugger/resources/app.asar/js/RNDebuggerWorker.js", origin: "file://", protocol: "file:", host: "", hostname: ""…}hash: ""host: ""hostname: ""href: "file:///opt/react-native-debugger/resources/app.asar/js/RNDebuggerWorker.js"origin: "file://"pathname: "/opt/react-native-debugger/resources/app.asar/js/RNDebuggerWorker.js"port: ""protocol: "file:"search: ""__proto__: WorkerLocation
```

`DEV_SERVER_ORIGIN` ends up being undefined.

This change fixes the issue: https://github.com/hedgepigdaniel/haul/commit/b4b464a6352b9219a10776e566f70c219d601063 but I'm not sure of the reason for using `window.location`

**Comment by hedgepigdaniel** [comment]
Bump, this is still the case with 1.0.0-rc.5

**Comment by stevefan1999** [comment]
https://github.com/callstack/haul/pull/464

**Comment by hedgepigdaniel** [comment]
Fixed by #462
